### PR TITLE
[CLEANUP] Always use anonymous functions for callables

### PR DESCRIPTION
**Pull request by oliverklee**
This makes it possible to have type declarations and annotations,
both helping static code analysis as well as making the code easier
to understand.

Also make the type annotations of the methods called from the
anonymous functions more specific.

### CODE REVIEW

File: psalm.baseline.xml [hunk 1]
```diff
@@ -8,7 +8,8 @@
       <code>$this-&gt;cssSelectorConverter === null</code>
       <code>$this-&gt;matchingUninlinableCssRules === null</code>
     </DocblockTypeContradiction>
-    <InvalidArgument occurrences="1">
+    <InvalidArgument occurrences="3">
+      <code>$cssRule</code>
       <code>$cssRule</code>
     </InvalidArgument>
```
Comment (JakeQZ) [inline psalm.baseline.xml:13]: `occurrences="3"` but there are only 2 in the list - ?

File: psalm.baseline.xml [hunk 2]
```diff
@@ -16,12 +17,8 @@
     <LessSpecificReturnStatement occurrences="1">
       <code>$excludedNodes</code>
     </LessSpecificReturnStatement>
-    <MixedArgument occurrences="1">
-      <code>$m[0]</code>
-    </MixedArgument>
     <MixedArgumentTypeCoercion occurrences="2">
       <code>$attributeName</code>
-      <code>\usort($cssRules['inlinable'], [$this, 'sortBySelectorPrecedence'])</code>
```
Comment (JakeQZ) [inline psalm.baseline.xml:24]: The `occurrences="2"` attribute has not changed but now there's only one occurrence.

File: src/CssInliner.php [hunk 3]
```diff
@@ -690,7 +691,16 @@ private function parseCssRules(string $css): array
             }
         }
 
-        \usort($cssRules['inlinable'], [$this, 'sortBySelectorPrecedence']);
+        \usort(
+            $cssRules['inlinable'],
+            /**
+             * @param array<string, string> $a
+             * @param array<string, string> $b
+             */
```
Comment (JakeQZ) [inline src/CssInliner.php:699]: I think the existing DocBlock type hint (`string[]`) for `sortBySelectorPrecedence` parameters was incorrect.  In full they should be `array{media: string, selector: string, hasUnmatchablePseudo: bool, declarationsBlock: string, line: int}` but probably only the keys used by the function need to be specified, i.e. `array{selector: string, line: int}`.

File: src/CssInliner.php [hunk 4]
```diff
@@ -745,8 +755,8 @@ private function selectorPartHasUnsupportedOfTypePseudoClass(string $selectorPar
     }
 
     /**
-     * @param string[] $a
-     * @param string[] $b
+     * @param array<string, string> $a
+     * @param array<string, string> $b
```
Comment (JakeQZ) [inline src/CssInliner.php:759]: Ditto.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
I think an existing DocBlock was wrong, so should be corrected with this change.  Also some `occurrences` attributes in `psalm.baseline.xml` now seem to mismatch with the actual count of `<code>` elements.

File: psalm.baseline.xml (hunk 1 above)
Comment (oliverklee) [reply psalm.baseline.xml:13]: Yes, that's strange indeed. Still, this is what Psalm generates (for whatever reason), and I'd like to keep it so we don't need to always change this manually when re-generating the file.

File: psalm.baseline.xml (hunk 2 above)
Comment (oliverklee) [reply psalm.baseline.xml:24]: Ditto. I've re-generated the file.

Thinking of this: I'll try if deleting if before re-generating it helps.

File: psalm.baseline.xml (hunk 1 above)
Comment (oliverklee) [reply psalm.baseline.xml:13]: No, this doesn'tt change anything.

File: psalm.baseline.xml (hunk 2 above)
Comment (oliverklee) [reply psalm.baseline.xml:24]: No, this doesn't change anything.

File: src/CssInliner.php (hunk 3 above)
Comment (oliverklee) [reply src/CssInliner.php]: Fixed and repushed (both instances).

File: psalm.baseline.xml (hunk 1 above)
Comment (JakeQZ) [reply psalm.baseline.xml:13]: OK.

File: psalm.baseline.xml (hunk 2 above)
Comment (JakeQZ) [reply psalm.baseline.xml:24]: Weird.  Looks like on the regeneration this one has been corrected, but different incorrect one has now arisen.

**Review by JakeQZ — APPROVED** [review]

**Pull request merged**